app/controllers/api/v1/base_controller.rb

```diff
@@ -8,11 +8,12 @@ class BaseController < ApplicationController
       private
 
       def require_authentication!
-        user = authenticate_or_request_with_http_token do |token, _options|
-          User.find_by(authentication_token: token)
-        end
+        user = User.find(request.headers["X-User-Id"])
 
-        if user.present?
+        if user.present? && Devise.secure_compare(
+          "Token #{user.authentication_token}",
+          request.headers["Authorization"]
+        )
```

Comment:
could this one be extracted to gain a nicer name?
like... .`token_valid?` or the like?
I am not a fan of stretching an if over multiple lines 🤔